private/pkg/manifest/module.go

```diff
@@ -107,6 +113,21 @@ func (b *memoryBlob) EqualContent(ctx context.Context, other Blob) (_ bool, retE
 	return true, nil
 }
 
+// AsProtoBlob returns the current blob as a proto module blob.
+func (b *memoryBlob) AsProtoBlob() (*modulev1alpha1.Blob, error) {
+	hashKind, ok := digestTypeToHashKind[b.digest.Type()]
+	if !ok {
+		return nil, fmt.Errorf("digest type %q not supported by module proto", b.Digest().Type())
+	}
+	return &modulev1alpha1.Blob{
+		Hash: &modulev1alpha1.Hash{
+			Kind:   hashKind,
+			Digest: b.digest.Bytes(),
+		},
+		Content: b.content,
+	}, nil
+}
+
```

Comment:
I'm a little weary about adding mappers for an internal type. It ties it to a representation it doesn't necessarily care about. Imagine what a "fileBlob" or a "GCPBlob" implementation would be; they all would look quite similar.

```go
func AsProtoBlob(ctx contentex.Context, blob Blob) (*modulev1alpha1.Blob, error) {
	file, err := blob.Open(ctx)
	if err != nil {
		return nil, err
	}
	content, err := io.ReadAll(file)
	if err != nil {
		return nil, err
	}
	blob := &modulev1alpha1.Blob{
		Hash: &modulev1alpha1.Hash{
			Kind:   hashKind,
			Digest: blob.Digest().Bytes(),
		},
		Content: content,
	}
	if err := file.Close(); err != nil {
		return nil, err
	}
	return blob, nil
}
```

Reply:
thx! done